scripts/bench_literals.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"mvdan.cc/garble/internal/literals"
+)
+
+const (
+	minDataLen      = 8
+	maxDataLen      = 64
+	stepDataLen     = 4
+	dataCountPerLen = 10
+	moduleName      = "test/literals"
+	garbleSeed      = "o9WDTZ4CN4w"
+
+	// For benchmarking individual obfuscators, we use package=obfuscator mapping
+	// and add a prefix to package name to make sure there are no collisions with system packages.
+	packagePrefix = "literals_bench_"
+)
+
+func generateRunSrc() string {
+	var sb strings.Builder
+
+	sb.WriteString(`
+var alwaysFalseFlag = false
+
+func noop(i interface{}) {
+	if alwaysFalseFlag {
+		println(i)
+	}	
+}
+
+func Run() {
+`)
+
+	dataStr := strings.Repeat("X", maxDataLen)
+	dataBytes := make([]string, maxDataLen)
+	for i := 0; i < len(dataBytes); i++ {
+		dataBytes[i] = strconv.Itoa(i)
+	}
+
+	for dataLen := minDataLen; dataLen <= maxDataLen; dataLen += stepDataLen {
+		for y := 0; y < dataCountPerLen; y++ {
+			fmt.Fprintf(&sb, "\tnoop(%q)\n", dataStr[:dataLen])
+			fmt.Fprintf(&sb, "\tnoop([]byte{%s})\n", strings.Join(dataBytes[:dataLen], ", "))
+		}
+	}
+
+	sb.WriteString("}\n")
+	return sb.String()
+}
+
+func buildTestGarble(tdir string) string {
+	garbleBin := filepath.Join(tdir, "garble")
+	if runtime.GOOS == "windows" {
+		garbleBin += ".exe"
+	}
+
+	output, err := exec.Command("go", "build", "-tags", "garble_testing", "-o="+garbleBin).CombinedOutput()
+	if err != nil {
+		log.Fatalf("garble build failed: %v\n%s", err, string(output))
+	}
+
+	return garbleBin
+}
+
+func writeDateFile(tdir, obfName, src string) error {
+	pkgName := packagePrefix + obfName
+
+	dir := path.Join(tdir, pkgName)
+	if err := os.MkdirAll(dir, 0o777); err != nil {
+		return err
+	}
+
+	f, err := os.Create(path.Join(dir, "data.go"))
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	fmt.Fprintf(f, "package %s\n\n", pkgName)
+	f.WriteString(src)
```

Comment:
nit: all the write calls to the file should check for errors.

Reply:
`Fprintf` too? 
I understand motivation, but it makes checking cumbersome:
```
_, err := f.WriteString(src)
handle(err)
```

But i can make specialized version of `handle` with additional parameter

Reply:
technically yes, any writes could fail. if you want to avoid checking multiple errors, you could also write into a strings.Builder, then use os.WriteFile. as long as the buffer doesn't get too big (like tens of megabytes), it's fine for a code generator like this.